File: src/main/java/com/aladdinworks5/service/CapacityAlertService.java
package com.aladdinworks5.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.aladdinworks5.domain.CapacityAlert;
import com.aladdinworks5.dto.CapacityAlertDTO;
import com.aladdinworks5.dto.CapacityAlertSearchDTO;
import com.aladdinworks5.dto.CapacityAlertPageDTO;
import com.aladdinworks5.dto.CapacityAlertConvertCriteriaDTO;
import com.aladdinworks5.service.GenericService;
import com.aladdinworks5.dto.common.RequestDTO;
import com.aladdinworks5.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface CapacityAlertService extends GenericService<CapacityAlert, Integer> {

	List<CapacityAlert> findAll();

	ResultDTO addCapacityAlert(CapacityAlertDTO capacityAlertDTO, RequestDTO requestDTO);

	ResultDTO updateCapacityAlert(CapacityAlertDTO capacityAlertDTO, RequestDTO requestDTO);

    Page<CapacityAlert> getAllCapacityAlerts(Pageable pageable);

    Page<CapacityAlert> getAllCapacityAlerts(Specification<CapacityAlert> spec, Pageable pageable);

	ResponseEntity<CapacityAlertPageDTO> getCapacityAlerts(CapacityAlertSearchDTO capacityAlertSearchDTO);
	
	List<CapacityAlertDTO> convertCapacityAlertsToCapacityAlertDTOs(List<CapacityAlert> capacityAlerts, CapacityAlertConvertCriteriaDTO convertCriteria);

	CapacityAlertDTO getCapacityAlertDTOById(Integer capacityAlertId);







}

File: aladdinworks5_ui/src/services/CapacityAlertService.js
import http from "../http-common"; 

class CapacityAlertService {
  getAllCapacityAlerts(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/capacityAlert/capacityAlerts`, searchDTO);
  }

  get(capacityAlertId) {
    return this.getRequest(`/capacityAlert/${capacityAlertId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/capacityAlert?field=${matchData}`, null);
  }

  addCapacityAlert(data) {
    return http.post("/capacityAlert/addCapacityAlert", data);
  }

  update(data) {
  	return http.post("/capacityAlert/updateCapacityAlert", data);
  }
  
  uploadImage(data,capacityAlertId) {
  	return http.postForm("/capacityAlert/uploadImage/"+capacityAlertId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new CapacityAlertService();
